core/src/kura.rs

```diff
@@ -220,36 +193,28 @@ impl Kura {
     }
 
     #[allow(clippy::expect_used, clippy::cognitive_complexity, clippy::panic)]
-    fn kura_recieve_blocks_thread(
-        weak_kura_reference: &Weak<Kura>,
+    fn kura_recieve_blocks_loop(
+        kura: &Kura,
         mut shutdown_receiver: tokio::sync::oneshot::Receiver<()>,
     ) {
+        let mut is_closed = false;
         loop {
-            // Here we essentially check if the thread should keep running.
-            // If all other references to `Kura` have been dropped, then
-            // the thread associated with that `Kura` instance should exit.
-            let strong_kura_reference = if let Some(kura_arc) = weak_kura_reference.upgrade() {
-                kura_arc
-            } else {
-                info!("Kura block thread is shutting down");
-                return;
-            };
+            let mut block_reciever_guard = kura
+                .block_reciever
+                .lock()
+                .expect("able to lock kura recieve block mutex");
 
+            // If kura receive shutdown than close block channel and write remaining blocks to the storage
             if shutdown_receiver.try_recv().is_ok() {
                 info!("Kura block thread is being forcibly shut down");
```

Comment:
Why is it *fordcibly*? Is there another way to shutdown?

Reply:
Without panic case, of course

Reply:
There was until very recently so this should probably just say `Kura block thread is being shut down`.